include/GoogleSync/GoogleSync.php

```diff
@@ -46,1088 +53,249 @@
-                $sReminder->timer_popup = $override->getMinutes() * 60;
-                $sReminder->related_event_module = $event_local->module_name;
-                $sReminder->related_event_module_id = $event_local->id;
-                $reminderId = $sReminder->save(false);
+        // First, we look for SuiteCRM meetings that are not on Google
+        foreach ($meetings as $meeting) {
+            $gevent = null;
+            if ( !empty($meeting->gsync_id) ) {
+                $gevent = $this->getGoogleEventById($meeting->gsync_id);
+            }
+            
+            $action = $this->pushPullSkip($meeting, $gevent);
+            $actionResult = $this->doAction($action, $meeting, $gevent);
 
-                $reminderInvitee = new Reminder_Invitee;
-                $reminderInvitee->reminder_id = $reminderId;
-                $reminderInvitee->related_invitee_module = 'Users';
-                $reminderInvitee->related_invitee_module_id = $this->workingUser->id;
-                $reminderInvitee->save(false);
+            if (!$actionResult) {
+                // TODO: confusing info log: $actionResult evaluated as string but it also false if converted to boolean. There is an empty string in log as result. - TODO: Inform the caller about this problem (exception?)
+                $this->logger->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - doAction Returned: ' . $actionResult); 
             }
         }
-        return  $event_local;
-    }
 
-    /**
-     * Create SuiteCRM Meeting event
-     *
-     * @param \Google_Service_Calendar_Event $event_remote The Google_Service_Calendar_Event we're creating a SuiteCRM Meeting for
-     *
-     * @return Meeting SuiteCRM Meeting Bean
-     */
-    public function createSuitecrmMeetingEvent($event_remote)
-    {
-        $GLOBALS['log']->debug(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Creating New SuiteCRM Meeting');
-        $meeting = new Meeting;
-        $meeting->id = create_guid();
-        $meeting->new_with_id = true;
-        $event_local = $this->updateSuitecrmMeetingEvent($meeting, $event_remote);
-        return $event_local;
-    }
-
-    /**
-     * Update Google Calendar Event from SuiteCRM Meeting
-     *
-     * @param Meeting $event_local SuiteCRM Meeting Bean
-     * @param \Google_Service_Calendar_Event $event_remote Google Event Object
-     *
-     * @return \Google_Service_Calendar_Event
-     */
-    public function updateGoogleCalendarEvent($event_local, $event_remote)
-    {
-
-        if ((!isset($event_local) || empty($event_local)) || (!isset($event_remote) || empty($event_remote))) {
-            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'ERROR:Missing Variables');
+        // Now, we look at the Google Calendar
+        $googleEvents = $this->getUserGoogleEvents();
+        if (!isset($googleEvents)) { // TODO: if condition never be true, as $googleEvents is definitely set just one line before.
```

Comment:
@gymad 
More cruft. this was used to catch if getUserGoogleEvents() returned null. I'll clean it up.

Reply:
Done